AK/Stack.h

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2021, Jesse Buhagiar <jooster669@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Vector.h>
+
+namespace AK {
+
+template<typename T, size_t stack_size>
+class Stack {
+public:
+    Stack() = default;
+    ~Stack() = default;
+
+    bool push(const T& item)
+    {
+        if (m_stack.size() >= stack_size)
+            return false;
+
+        m_stack.append(item);
+        return true;
+    }
+
+    bool push(T&& item)
+    {
+        if (m_stack.size() >= stack_size)
+            return false;
+
+        m_stack.append(move(item));
+        return true;
+    }
+
+    bool is_empty() const
+    {
+        return m_stack.is_empty();
+    }
+
+    size_t size() const
+    {
+        return m_stack.size();
+    }
+
+    bool pop()
+    {
+        if (is_empty())
+            return false;
+
+        m_stack.resize_and_keep_capacity(m_stack.size() - 1);
+        return true;
+    }
+
+    T& top()
+    {
+        return m_stack.last();
+    }
+
+    const T& top() const
+    {
+        return m_stack.last();
+    }
+
+private:
+    Vector<T, stack_size> m_stack;
```

Comment:
Maybe I'm being silly, but can we instead switch this to fixed storage?
Vector has some mystifying behaviours and errors:
> what if you wanted to write code, but the compiler said
```c++
../.././AK/Vector.h:60:34: error: '*(AK::Vector<Wasm::Value, 0>*)((char*)&<unnamed> + offsetof(AK::EntryType, AK::Variant<Wasm::Value, Wasm::Label, Wasm::Frame>::m_data[4])).AK::Vector<Wasm::Value>::m_outline_buffer' may be used uninitialized [-Werror=maybe-uninitialized]
   60 |         , m_outline_buffer(other.m_outline_buffer)
      |                            ~~~~~~^~~~~~~~~~~~~~~~
```

> and then you checked and saw
```
T* m_outline_buffer { nullptr };
```

Reply:
As per Discord conversation, this is a refactor for another time :)